src/linker/Linker.Dataflow/ReflectionMethodBodyScanner.cs

```diff
@@ -654,6 +731,29 @@ public override bool HandleCall (MethodBody callingMethodBody, MethodReference c
 				calledMethodDefinition,
 				operation);
 
+			if (calledMethodDefinition.IsPInvokeImpl) {
+				// Is the PInvoke dangerous?
+				bool comDangerousMethod = IsComInterop (calledMethodDefinition.MethodReturnType, calledMethodDefinition.ReturnType);
+				foreach (ParameterDefinition pd in calledMethodDefinition.Parameters) {
+					TypeReference paramTypeReference = pd.ParameterType;
+					if (paramTypeReference is TypeSpecification) {
+						paramTypeReference = (paramTypeReference as TypeSpecification).ElementType;
```

Comment:
I see that this is just copying existing logic from MarkStep, but I don't understand why we look at the element type here. Is this again to handle modifiers? Maybe it would be better to handle this kind of logic inside of the `IsComInterop` helper.

Reply:
I think it helps us to deal with arrays, but yes, it might be better to unwrap that in IsComInterop helper.

I was scratching my head why I wrote it that way but this code has been there for 4 years, I just failed to notice that paramTypeDefinition is not actually the parameter type definition.

Reply:
Removed the spec check since Resolve in the `IsComInterop `helper seems to handle this